src/Rector/MethodCall/WhereToWhereLikeRector.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RectorLaravel\Rector\MethodCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name;
+use PhpParser\Node\Scalar\String_;
+use PHPStan\Type\ObjectType;
+use Rector\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Webmozart\Assert\Assert;
+
+/**
+ * @see https://github.com/laravel/framework/pull/52147
+ * @see \RectorLaravel\Tests\Rector\MethodCall\WhereToWhereLikeRector\WhereToWhereLikeRectorTest
+ * @see \RectorLaravel\Tests\Rector\MethodCall\WhereToWhereLikeRector\WhereToWhereLikeRectorPostgresTest
+ */
+final class WhereToWhereLikeRector extends AbstractRector implements ConfigurableRectorInterface
+{
+    private const array WHERE_LIKE_METHODS = [
+        'where' => 'whereLike',
+        'orwhere' => 'orWhereLike',
+    ];
+
+    private bool $usingPostgresDriver = false;
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Changes `where` method calls to `whereLike` method calls in the Eloquent & Query Builder', [
+                new ConfiguredCodeSample(
+                    <<<'CODE_SAMPLE'
+$query->where('name', 'like', 'Rector');
+$query->orWhere('name', 'like', 'Rector');
+$query->where('name', 'like binary', 'Rector');
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+$query->whereLike('name', 'Rector');
+$query->orWhereLike('name', 'Rector');
+$query->whereLike('name', 'Rector', true);
+CODE_SAMPLE
+                    ,
+                    ['usingPostgresDriver' => false]
+                ),
+                new ConfiguredCodeSample(
+                    <<<'CODE_SAMPLE'
+$query->where('name', 'ilike', 'Rector');
+$query->orWhere('name', 'ilike', 'Rector');
+$query->where('name', 'like', 'Rector');
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+$query->whereLike('name', 'Rector');
+$query->orWhereLike('name', 'Rector');
+$query->whereLike('name', 'Rector', true);
+CODE_SAMPLE
+                    ,
+                    ['usingPostgresDriver' => true]
+                ),
+            ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
```

Comment:
This could be expanded to Static calls for a Model as well, as `Model::where('a', 'like', 'b')` wouldn't be covered by this rule as it uses a proxy system.